style-guide.md

```diff
@@ -0,0 +1,165 @@
+# edX JavaScript Conventions
+
+# TODO
+
+I'm looking for widespread agreement on everything contained in this document!
+In particular, I'd love feedback on the [Goals](#goals), [Style](#style), and
+[Conventions](#conventions) sections.
+
+Here are a few things that would be awesome to have:
+- [ ] define and link to a jshint/jscs/eslint config
```

Comment:
:thumbsup: - Andy and I are thinking about taking this on for the hackathon. Most of the recommendations in this styleguide are compatible with the Airbnb ESLint config, which I like a lot.